### ISSUE TITLE
Could not load file or assembly 'System.Management.Automation

### ISSUE THREAD

**Issue by erenturkm** [PowerShell/PowerShell#13027]
I need to have Single Executable including Powershell SDK.
1) When I run through the debugger, I dont see any Exceptions.
2) When I build and publish in Debug configuration, I dont see any Exceptions.
3) This only happens in 7.0.2, I dont see any exceptions in 7.0.0

## Steps to reproduce
When I build and publish in Release configuration, I receive an Exception immediatly when I enter the function that I call powershell SDK commands.  Below is the Exception Message:

Could not load file or assembly 'System.Management.Automation, Version=7.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The located assembly's manifest definition does not match the assembly reference. (0x80131040),   at SystemTools.Shell.PowerShellSDKExecutor.RunCommands(List`1 cmds)

## Environment data

```
PSVersion=7.0.2
PSEdition=Core
WSManStackVersion=3.0
PSRemotingProtocolVersion=2.3
```

**Comment by iSazonov** [comment]
Can you share a code of the simple application to reproduce the issue?

**Comment by adityapatwardhan** [comment]
@erenturkm Please have a look at: https://devblogs.microsoft.com/powershell/depending-on-the-right-powershell-nuget-package-in-your-net-project/ 

It is an excellent blog about using the various NuGet packages we publish.

**Comment by rjmholt** [comment]
Sounds like your application is trying to load 7.0.2 but is finding a different version. How are you building it? Can you also try deleting the obj and bin directories and rebuilding it cleanly?

**Comment by erenturkm** [comment]
Hi,

Thanks for the suggestions. I have been using the 7.0.0 library for sometime and works like a charm. (Yes I did read and selected according to the mentioned article). I tested with both 7.0.2 and 7.0.0 and saw it only on 7.0.2. It looks like it is related to the linking stage and not the compilation and only for relase configuration.      

I am not sure what changed (Yes I regulary clean obj and bin directories) but I can not reproduce the issue anymore. I will come back when it resurfaces.
